src/integration/test_suites/cptra_ss_i3c_recovery/cptra_ss_i3c_recovery.c

```diff
@@ -0,0 +1,403 @@
+// Description: I3C Smoke test for Caliptra Subsystem
+// Author     : Nilesh Patel
+// Created    : 2025-01-14
+// Comments   : 
+//  This is a smoke test for I3C interface on Caliptra. 
+//  The test brings up the I3C interface and sends a command to the I3C device. 
+//  The test is expected to pass if the I3C device responds with the expected data.
+
+#include "soc_address_map.h"
+#include "printf.h"
+#include "riscv_hw_if.h"
+#include "soc_ifc.h"
+#include "string.h"
+#include "stdint.h"
+
+#define STATUS_CHECK_LOOP_COUNT_FOR_RECOVERY 20
+
+volatile char* stdout = (char *)0x21000410;
+// volatile char* stdout = (char *)0xd0580000;
+
+#ifdef CPT_VERBOSITY
+    enum printf_verbosity verbosity_g = CPT_VERBOSITY;
+#else
+    enum printf_verbosity verbosity_g = LOW;
+#endif
+
+
+// -- function to update various register to default values
+// PROT_CAP, DEVICE_ID, HW_STATUS
+void boot_i3c_recovery_reg(void) {
+
+    uint32_t i3c_reg_data = 0x00000001;
+
+    //-- PROT_CAP
+    VPRINTF(LOW, "MCU: Updating I3C Recovery Registers\n");
+    lsu_write_32( SOC_I3CCSR_I3C_EC_SECFWRECOVERYIF_PROT_CAP_0, i3c_reg_data++);
+    lsu_write_32( SOC_I3CCSR_I3C_EC_SECFWRECOVERYIF_PROT_CAP_1, i3c_reg_data++);
+    lsu_write_32( SOC_I3CCSR_I3C_EC_SECFWRECOVERYIF_PROT_CAP_2, i3c_reg_data++);
+    lsu_write_32( SOC_I3CCSR_I3C_EC_SECFWRECOVERYIF_PROT_CAP_3, i3c_reg_data++);
+
+    //-- DEVICE_ID
+    lsu_write_32( SOC_I3CCSR_I3C_EC_SECFWRECOVERYIF_DEVICE_ID_0, i3c_reg_data++);
+    lsu_write_32( SOC_I3CCSR_I3C_EC_SECFWRECOVERYIF_DEVICE_ID_1, i3c_reg_data++);
+    lsu_write_32( SOC_I3CCSR_I3C_EC_SECFWRECOVERYIF_DEVICE_ID_2, i3c_reg_data++);
+    lsu_write_32( SOC_I3CCSR_I3C_EC_SECFWRECOVERYIF_DEVICE_ID_3, i3c_reg_data++);
+    lsu_write_32( SOC_I3CCSR_I3C_EC_SECFWRECOVERYIF_DEVICE_ID_4, i3c_reg_data++);
+    lsu_write_32( SOC_I3CCSR_I3C_EC_SECFWRECOVERYIF_DEVICE_ID_5, i3c_reg_data++);
+    lsu_write_32( SOC_I3CCSR_I3C_EC_SECFWRECOVERYIF_DEVICE_ID_6, i3c_reg_data++);
+
+    //-- HW_STATUS
+    lsu_write_32( SOC_I3CCSR_I3C_EC_SECFWRECOVERYIF_HW_STATUS, i3c_reg_data++);
```

Comment:
Seems we are only setting the lower bits of these registers. Should we test the upper bits?  Let setting this to 0x1111_1111, 0x2222_2222, or is that not the intent of this test?